src/common/ufo-ctlab.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2017 Karlsruhe Institute of Technology
+ *
+ * This file is part of Ufo.
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef UFO_CTLAB_H
+#define UFO_CTLAB_H
+
+#include <ufo/ufo.h>
+#include "ufo-scarray.h"
+
+typedef struct {
+    UfoScarray *x;
+    UfoScarray *y;
+    UfoScarray *z;
+} UfoPoint;
+
+typedef struct {
+    UfoPoint *position;
+    UfoPoint *angle;
+} UfoVector;
+
+typedef struct {
+    UfoPoint  *source_position;
+    UfoPoint  *volume_angle;
+    UfoVector *axis;
+    UfoVector *detector;
+} UfoCTlab;
```

Comment:
These structures don't follow the "private data" convention but they prevent telescoping functions to get/set individual elements, e.g. `ufo_ctlab_set_axis_position_x (UfoCTlab *lab, UfoScarray *array)`. The "real data", i.e. the `GValueArray` in the underlying `UfoScarray` attributes of course remain private. This seemed to me like a good compromise. What does you gut say, @matze?

Reply:
I think that's fine as long as this does not need to be exposed publicly. But I have one remark: could we change CTLab to something else and more descriptive?

Reply:
And if it should be public how to do it? The number of accessor functions would be 18 and that's just getters.
Regarding the name, hiw does `CTGeometry` sound?

Reply:
> And if it should be public how to do it?

Well, definitely not with C structs. How should the Python user access them?

> Regarding the name, hiw does CTGeometry sound?

The question is what the name is trying to convey? Would `LTGeometry` or `ConebeamGeometry` exist as well?

Reply:
Yeah that wouldn't work for sure. I wasn't thinking in terms of glib and multiple programming languages. I was just wondering how could one do it in plain C.
Regarding the name, it actually reflects laboratory geometry, hence the first name. You have a source somewhere, your axis and a detector. These three things enable you to do tomo/lamino with cone/parallel beam. There will not be any `LTGeometry` because what we have now is general enough. My inspiration came from our CT lab, which is also capable of doing laminography.

Reply:
Okay, so with public I meant *really* public not internal to the plugins, so I think if that's the case then it's fine. Concerning the name, then I'd really prefer geometry or setup or some such and avoid names that might not mean a thing to other people.

Reply:
Hmm, maybe setup is also not that bad. Let me eat on it and I will just push something :smile:

Reply:
I will go with `CTGeometry`.